jina/serve/runtimes/worker/http_fastapi_app.py

```diff
@@ -70,6 +98,8 @@ async def post(body: input_model, response: Response):
             req.parameters = body.parameters
             req.header.exec_endpoint = endpoint_path
             resp = await caller(req)
+            if request_models_map[endpoint_path]['is_generator']:
```

Comment:
this `is_generator`  flag should be computed outside the `post` so that u do not do this access here